### PR TITLE
Test coverage

### LINKED ISSUE #9: Setup Coveralls
For code coverage reporting!

### PR DESCRIPTION
**Pull request by manuelurenah**
### Description

* Adds remaining test cases to up the code coverage to 100%
* Minor modifications to add circleCI integration
* Adds coveralls dependency
* Closes #9

### CODE REVIEW

File: services/storage.test.js [hunk 1]
```diff
@@ -2,24 +2,22 @@
...
 
     lib.setDb(fakeDb);
   });
 
   describe('get', function () {
     const fn = lib[this.description];
 
-    it('should return a value by id', function () {
+    it('calls the get method from the db service', function () {
       fakeDb.get.mockResolvedValue({ id: 'foo', username: 'Pizza', provider: 'google' });
```
Comment (jjpaulino) [inline services/storage.test.js]: ⭐️

File: services/storage.test.js [hunk 2]
```diff
@@ -32,7 +30,7 @@ describe(_startCase(filename), function () {
   describe('put', function () {
     const fn = lib[this.description];
 
-    it('should update an item in the db', function () {
+    it('calls the update method from the db service', function () {
       const data = {
         username: 'Pizza',
```
Comment (jjpaulino) [inline services/storage.test.js]: GUOAT: Greatest Username Of All Time

File: utils.test.js [hunk 3]
```diff
@@ -53,4 +94,250 @@ describe(_startCase(filename), function () {
...
+  });
+
+  describe('getUri', function () {
+    const fn = lib[this.description];
+
+    it('returns a formatted uri', function () {
+      const req = {
+        hostname: 'foo.com',
+        baseUrl: '/bar',
+        path: '/pizza'
```
Comment (jjpaulino) [inline utils.test.js]: This is the `real` path to happiness!

File: services/responses.js [hunk 4]
```diff
@@ -130,11 +130,7 @@ function sendTextErrorCode(code, message, res) {
  * @param {object} res
  */
 function notFound(err, res) {
-  if (!(err instanceof Error) && err) {
-    res = err;
-  }
-
-  const message = 'Not Found',
+  const message = err.message,
```
Comment (fxisco) [inline services/responses.js]: should we add a fallback or maybe check if `err` exists?

File: services/responses.test.js [hunk 5]
```diff
@@ -327,4 +336,123 @@ describe(_startCase(filename), function () {
...
+          provider: 'bar',
+          auth: 'baz'
+        };
+
+      fakeDb.get.mockResolvedValue(mockData);
+
+      fn(req, mockRes);
+
+      expect(fakeDb.get).toBeCalledWith(req.uri);
+      done();
```
Comment (fxisco) [inline services/responses.test.js]: this done is different from the afterEach call?

File: index.test.js [hunk 6]
```diff
@@ -148,4 +149,53 @@ describe(_startCase(filename), function () {
...
+      expect(lib({})).toEqual([]);
+    });
+
+    it('should set required middlewares', function () {
+      strategyService.createStrategy = jest.fn();
+      strategyService.addAuthRoutes = jest.fn();
+
+      lib(options);
+
+      expect(router.use).toBeCalledTimes(7);
```
Comment (james-owen) [inline index.test.js]: For these `toBeCalledTimes()` it might be helpful to comment what's calling it. What do you think?

File: utils.test.js [hunk 7]
```diff
@@ -53,4 +94,250 @@ describe(_startCase(filename), function () {
...
+
+      expect(results.length).toEqual(1);
+      expect(results[0].url).toEqual(expected[0].url);
+    });
+  });
+
+  describe('verify', function () {
+    const fn = lib[this.description];
+
+    it('throws if there is no username field', function () {
```
Comment (fxisco) [inline utils.test.js]: What do you think about using a beforeEach function to group?

```js
properties = { username: 'username' };
profile = { username: '' }, // maybe
req = {},
next = jest.fn(),
```

And just do modifications in the tests you want to have different situations

File: services/responses.test.js [hunk 8]
```diff
@@ -327,4 +336,123 @@ describe(_startCase(filename), function () {
       fn()(req, mockRes);
     });
   });
+
+  describe('getRouteFromDB', function () {
+    const fn = lib[this.description];
+
+    afterEach(function (done) {
+      return fakeDb.clearMem().then(done);
```
Comment (PedroAlbR) [inline services/responses.test.js]: Is `then` needed here? I don't see it in this line so maybe we could make them look the same https://github.com/clay/amphora-auth/pull/14/files#diff-fde8b926a5441888c2c16db0c4d1046bR308

**Pull request merged**